guardian/recipes-backend
# Rewrite featured and preview images to fastly URLs on write, if present

## What does this change?

Rewrites image URLs for featured and preview images to URLs that use the [fastly resizer service](https://github.com/guardian/fastly-image-service/tree/34399e065bc85b8ca4d5adeeca02b4b6404eaa98), if it's possible (the image must be present and have a media and crop ID, or a URL from which it's possible to extract a crop ID.)

in the future, we'd like to add a `templateUrl` property to allow the app to template in its own parameters, but in the short term this should get us off the ground.

Only newly edited recipes will see this change.

## How to test

- The automated tests should pass. I've thought of a few unhappy paths – perhaps there are more. See the test descriptions for details.
- Deploy to CODE, and make a change to a recipe. You should see the appropriate fastly URLs for the recipe images in backend. They should work!

- [x] Tested in CODE with 1 ([composer](https://composer.code.dev-gutools.co.uk/content/65e5ce818f086762e6507c74), [recipe](https://recipes.code.dev-guardianapis.com/content/chTD37k2MCG8cXSSqKVbV3vpdwBXYCa9ec7hD-hGT6E)), 2 ([composer](https://composer.code.dev-gutools.co.uk/content/5a16b8e793c7ae06fe472272), [recipe](https://recipes.code.dev-guardianapis.com/content/iz4V62YerXHH3GgVx05lTpw_ffEVBPQERUTYvbM9Vko)) – note the differing image URLs. The recipe-backend images should all be i.guim.co.uk.

## How can we measure success?

Smaller images for the recipes app.
